app/controllers/v0/sessions_controller.rb

```diff
@@ -62,44 +68,28 @@ def async_create_evss_account(user)
 
     def handle_completed_slo
       logout_response = OneLogin::RubySaml::Logoutresponse.new(params[:SAMLResponse], saml_settings)
-      logout_request = SingleLogoutRequest.find(logout_response.in_response_to)
+      logout_request = SingleLogoutRequest.find(logout_response&.in_response_to)
 
-      logger.info "LogoutResponse is: #{logout_response}"
+      errors = []
+      errors.concat(logout_response.errors) unless logout_response.validate(true)
+      errors << "Logout Request not found! ID=#{logout_response&.in_response_to}" if logout_request.nil?
+      errors << 'Session not found!' if (session = Session.find(logout_request&.token)).nil?
```

Comment:
```ruby
session = Session.find(logout_request&.token)
user = User.find(session&.uuid)

# option 1 - using exists? more readable but makes a 2nd redis call
errors << 'Session not found!' unless Session.exists?(logout_request&.token)
errors << 'User not found!' unless User.exists?(session&.uuid)

# option 2 - use of nil? is DRY'er an
errors << 'Session not found!' if session.nil?
errors << 'User not found!' if user.nil?
```

I actually like option 2 better because it's DRY'er (doesnt repeat the lookup) and only makes one redis call instead of two.  `exists?` reads nicer, but I don't think it's worth it in this case.